### PR TITLE
ESM Baseline (WIP)

### PR DESCRIPTION
**Pull request by lapp0**
Done:
- Update model parameters to reflect ESM++ vocab_size https://huggingface.co/Synthyra/ESMplusplus_large/blob/main/modeling_esm_plusplus.py#L868-L874
- Remove causal mask

Questions:
- What is the max sequence length we should consider when validating? 2048?
- Other than validation loss, what eval metrics do you recommend (to run at end of training)

Remaining:
- validate with dataset

When you're ready with the dataloader, I can merge it into my branch and experiment. Happy to help on the data side.

### CODE REVIEW

**Comment by lhallee** [comment]
Thanks!

I'm processing the training data with appropriate splits, this test set should be a solid evaluation. Will run it on all the ESM2 models and track loss as well as sequence reconstruction metrics. I'll update the readme with these metrics soon.

We should evaluate on sequences up to 1024, as this was the maximum length trained on for ESM2. However, if we are able to get speed up training on longer sequences somehow that should be fine as long as we eval on 1024 max.

I will push what I have for data processing by the end of today.

**Comment by lapp0** [comment]
Great! Please ping me when https://github.com/Synthyra/SpeedRunningESM2/blob/master/train_gpt2.py#L402-L403 is updated and I'll take a whack at refining the implementations hyperparameters such that they're optimal for this task. (Also so I can ensure the attention masking is sane)

**Comment by lhallee** [comment]
Hey @lapp0!

- dataset is processed and on [Huggingface](https://huggingface.co/datasets/Synthyra/omg_prot50) with new benchmark test set
- metrics for ESM2 models posted in readme
- caching OMG prot50 works
- updated #L402-L403, now #L410-413

Haven't had time to test yet, will certainly need some tweaks.
